draft-ietf-privacypass-rate-limit-tokens.md

```diff
@@ -922,6 +957,40 @@ struct {
 } Token;
 ~~~
 
+## Penalization of Invalid Clients and Issuers {#penalization}
+
+When an Attester encounters incorrect and potentially malicious behavior
+by either Clients or Issuers, it needs to "penalize" them. This involves
+keeping state about invalid protocol events, and determining when a
+particular Client or Issuer has exceeded a threshold for such events.
+
+Each category of invalid events can have a different threshold, and can apply
+to Clients, Issuers, or both. Different penalization events are described
+below, with suggested thresholds. This list is not exhaustive.
+
+- Client Key changing more than once over any two consecutive policy windows.
+This is a Client-specific penalization event. A suggested threshold for
+penalizing the Client based on this is one event.
+
+- Issuer responses missing the "Sec-Token-Origin" header on a 2xx response.
+This is an Issuer-specific penalization event. A suggested threshold for
+penalizing the Issuer based on this is ten events across all Clients.
+
+- Anonymous Issuer Origin ID (from Issuer) collision across two Anonymous
+Client IDs for a particular Client within a single policy window. This
+penalization event is applicable for both Clients and Issuers. A suggested
+threshold for penalizing the Issuer is ten events across all Clients.
+A suggested threshold for penalizing the Client is two events across multiple
+Issuers, or five events with a single Issuer.
+
+Once a Client or Issuer passes the threshold for penalization, the Attester
+rejects future requests from that Client or for that Issuer. This SHOULD
```

Comment:
Is there some sort of indicator to the client that the issuer has been penalized? Or does it just appear to clients that they're no longer trusted?

Reply:
I don't think that signal is useful for the client to see since they don't trust the issuer by default anyway, but maybe there's a use case for it?

Reply:
I think this could be a bit proprietary — how the attester wants to reject clients beyond just an HTTP response code.